File: VB/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports Example.Models

Namespace Example.Controllers
	<HandleError> _
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			ViewData("Message") = "Welcome to DevExpress Extensions for ASP.NET MVC!"

			Return View()
		End Function

		Public Function GridViewPartial() As ActionResult
			Return PartialView("GridViewPartial")
		End Function

		Public Function GetRowData(ByVal keyValue As Int32) As JsonResult
			Dim result As New JsonResult()

			' get data by key value from you data source and put it to result.Data
			Dim context As New DataClassesDataContext()
			Dim product = context.Products.Select(Function(p) p).Where(Function(p) p.ProductID = keyValue).First()

			result.Data = New With {Key .Discontinued = product.Discontinued}

			Return result
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Example.Models;

namespace Example.Controllers {
    [HandleError]
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewData["Message"] = "Welcome to DevExpress Extensions for ASP.NET MVC!";

            return View();
        }

        public ActionResult GridViewPartial() {
            return PartialView("GridViewPartial");
        }

        public JsonResult GetRowData(Int32 keyValue) {
            JsonResult result = new JsonResult();

            // get data by key value from you data source and put it to result.Data
            DataClassesDataContext context = new DataClassesDataContext();
            var product = context.Products
                .Select(p => p)
                .Where(p => p.ProductID == keyValue)
                .First();

            result.Data = new { Discontinued = product.Discontinued };

            return result;
        }
    }
}
